core/templates/domain/skill/editable-skill-backend-api.service.ts

```diff
@@ -16,124 +16,164 @@
  * @fileoverview Service to send changes to a skill to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('EditableSkillBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_SKILL_DATA_URL_TEMPLATE', 'SKILL_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EDITABLE_SKILL_DATA_URL_TEMPLATE, SKILL_DATA_URL_TEMPLATE) {
-    var _fetchSkill = function(skillId, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skill = angular.copy(response.data.skill);
-        var groupedSkillSummaryDicts = angular.copy(
-          response.data.grouped_skill_summaries);
-        if (successCallback) {
-          successCallback({
-            skill: skill,
-            groupedSkillSummaries: groupedSkillSummaryDicts
-          });
-        }
-      }, function(errorResponse) {
+// TODO(#7165): Replace any with exact type.
+interface editableSkillResponseConfig{
+    skill?: any;
+    skills?: any;
+    // eslint-disable-next-line camelcase
+    grouped_skill_summaries?: any;
+  }
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableSkillBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  // TODO(#7165): Replace any with exact type.
+  _fetchSkill(
+      skillId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void
+  ): void {
+    const skillDataUrl = this.urlInterpolationService.interpolateUrl(
+      SkillDomainConstants.EDITABLE_SKILL_DATA_URL_TEMPLATE, {
+        skill_id: skillId
+      }
+    );
+    this.http.get<editableSkillResponseConfig>(skillDataUrl).toPromise().then(
```

Comment:
What's <editableSkillResponseConfig> doing here?

Reply:
I followed this: https://angular.io/guide/http#requesting-a-typed-response

Reply:
Do we really want this here and below? I guess it would work fine without this too, right?

Reply:
Yeah it will work fine without this too, I will remove them.

Reply:
Done!